pkg/toolchain/create_resource.go

```diff
@@ -0,0 +1,25 @@
+package toolchain
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CreateFromYAML creates a resource from a YAML manifest
+func CreateFromYAML(s *runtime.Scheme, cl client.Client, content []byte) error {
+	decoder := serializer.NewCodecFactory(s).UniversalDeserializer()
+	obj := unstructured.Unstructured{}
+	_, _, err := decoder.Decode(content, nil, &obj)
+	if err != nil {
+		return err
+	}
+	if err = cl.Create(context.TODO(), &obj); err != nil && !errors.IsAlreadyExists(err) {
```

Comment:
I guess that we don't care about updates for now.

Reply:
> I'm curious - why do we need this file at all? The spec is empty, so why couldn't we create the resource programmatically in go? Is it because of a better understanding of what is being created or is there something else behind it?
On the other hand, I guess that the Che installation file will be maybe more complicated - in that case it make sense for such a file... Then it would also make sense to do it for TektonInstallation file to keep consistency

yes, Che installation resource is slightly more complicated, and for the sake of consistency, it's better to have both in the same format. I believe that for all our CRs, YAML is better than programmatic/go.

> As for the dirs - yes, I would prefer one directory for both files - I have a feeling that we overuse the directories and packages (mainly in this project). As for the name - why not installation or installation-cr, but to be clear - I don't insist on it ;-)

ok, I'll move the code in `pkg/installation` and the YAML file in `deploy/installation` for the sake of consistency, too. At the end of the day, these packages will then contain the code for both Che and Tekton.

Reply:
please see the changes in https://github.com/codeready-toolchain/toolchain-operator/pull/16/commits/5c221dd401cd8549409c0dad692be61953ab8aac 

This commit brings the following changes:
- custom resources (.yaml files) in `deploy/installation/tekton` (and in the future `deploy/installation/che`
- all go code for tekton installation moved to `pkg/controller/tektoninstallation` (ie, no more `pkg/tekton` aside of `pkg/controller/tektoninstallation`). 

I'm not sure why we had both `pkg/controller/tektoninstallation` plus `pkg/tekton`, so please let me know if this regrouping causes troubles.